### PR TITLE
Block non-staff from using staff login

### LINKED ISSUE #201: Block non-staff from using staff login
Currently any gmail/g-apps account will work with the staff login button (they still won't have any staff permissions)

Ideas for fixing this:
- Check if the user has staff roles after logging in. If they don't, force log them out
- Somehow check their input email and make sure it has @hackillinois.org suffix (might be hard since it happens inside chrome?)

### PR DESCRIPTION
**Pull request by patrickfeltes**
Checks for staff role on JWT to make sure that non-staff members cannot use staff login with an arbitrary gmail account. Fixes #201

### CODE REVIEW

File: app/src/main/java/org/hackillinois/android/view/LoginActivity.kt [hunk 1]
```diff
@@ -70,15 +68,36 @@ class LoginActivity : AppCompatActivity() {
...
-                    runOnUiThread {
+
+                    if (getOAuthProvider() == "google") {
+                        api.roles().enqueue(object : Callback<Roles> {
+                            override fun onFailure(call: Call<Roles>, t: Throwable) {
+                                showFailedToLogin()
+                            }
+
+                            override fun onResponse(call: Call<Roles>, response: Response<Roles>) {
+                                if (response.isSuccessful) {
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/view/LoginActivity.kt]: && here?

File: app/src/main/java/org/hackillinois/android/view/LoginActivity.kt [hunk 2]
```diff
@@ -70,15 +68,36 @@ class LoginActivity : AppCompatActivity() {
...
+                                    if (response.body()?.roles?.contains("Staff") == true) {
+                                        JWTUtilities.writeJWT(applicationContext, it)
+                                        launchMainActivity()
+                                    }
+                                }
+                                showFailedToLogin()
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/view/LoginActivity.kt]: else {} around here for clarity?

File: app/src/main/java/org/hackillinois/android/view/LoginActivity.kt [hunk 3]
```diff
@@ -70,15 +68,36 @@ class LoginActivity : AppCompatActivity() {
                             Log.e("LoginActivity", "Notifications update timed out!")
                         }
                     }
-                    JWTUtilities.writeJWT(applicationContext, it)
-                    runOnUiThread {
+
+                    if (getOAuthProvider() == "google") {
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/view/LoginActivity.kt]: This if block could be its own function since it's so long

Comment (bstrauch24) [reply app/src/main/java/org/hackillinois/android/view/LoginActivity.kt]: There's a lot of nested blocks in this class in general

Comment (patrickfeltes) [reply app/src/main/java/org/hackillinois/android/view/LoginActivity.kt]: Yeah I'm hoping to remove the nested callbacks with a new Kotlin feature with Retrofit.

**Pull request merged**